firefox_puppeteer/api/software_update.py

```diff
@@ -0,0 +1,499 @@
+    """)
+
+
+class MARChannels(BaseLib):
+    """Class to handle the allowed MAR channels as listed in update-settings.ini."""
+    INI_SECTION = 'Settings'
+    INI_OPTION = 'ACCEPTED_MAR_CHANNEL_IDS'
+
+    def __init__(self, marionette_getter):
+        BaseLib.__init__(self, marionette_getter)
+
+        self.ini_file_path = self.marionette.execute_script("""
+          Components.utils.import('resource://gre/modules/Services.jsm');
+
+          let file = Services.dirsvc.get('GreD', Components.interfaces.nsIFile);
+          file.append('update-settings.ini');
+
+          return file.path;
+        """)
+
+    @property
+    def config_file_path(self):
+        """The path to the update-settings.ini file."""
+        return self.ini_file_path
+
+    @property
+    def config_file_contents(self):
+        """The contents of the update-settings.ini file."""
+        with open(self.ini_file_path) as f:
+            return f.read()
+
+    @property
+    def channels(self):
+        """The channels as found in the ACCEPTED_MAR_CHANNEL_IDS option
+        of the update-settings.ini file.
+
+        :returns: An array of channel names
+        """
+        channels = self.marionette.execute_script("""
+          Components.utils.import("resource://gre/modules/FileUtils.jsm");
+          let iniFactory = Cc['@mozilla.org/xpcom/ini-processor-factory;1']
+                           .getService(Components.interfaces.nsIINIParserFactory);
+
+          let file = new FileUtils.File(arguments[0]);
+          let parser = iniFactory.createINIParser(file);
+
+          return parser.getString(arguments[1], arguments[2]);
+        """, script_args=[self.ini_file_path, self.INI_SECTION, self.INI_OPTION])
+        return channels.split(',')
+
+    @channels.setter
+    def channels(self, channels):
+        """Set the channels in the update-settings.ini file.
+
+        :param channels: an array of channel names
+        """
+        new_channels = ','.join(channels)
+        self.marionette.execute_script("""
+          Components.utils.import("resource://gre/modules/FileUtils.jsm");
+          let iniFactory = Cc['@mozilla.org/xpcom/ini-processor-factory;1']
+                           .getService(Components.interfaces.nsIINIParserFactory);
+
+          let file = new FileUtils.File(arguments[0]);
+
+          let writer = iniFactory.createINIParser(file)
+                       .QueryInterface(Components.interfaces.nsIINIParserWriter);
+
+          writer.setString(arguments[1], arguments[2], arguments[3]);
+          writer.writeFile(null, Components.interfaces.nsIINIParserWriter.WRITE_UTF16);
+        """, script_args=[self.ini_file_path, self.INI_SECTION, self.INI_OPTION, new_channels])
+
+    def add_channels(self, channels):
+        """Add channels to the update-settings.ini file.
+
+        :param channels: an array of channel names to add
+        """
+        new_channels = set(self.channels) | set(channels)
+        self.channels = new_channels
```

Comment:
As mentioned in my last review cycle you can make `self.channels` to be a Set by default. So `channels` can return a Set instead of an array. That would make it easier across the file to use it. Maybe also assign it directly?